File: LinkERP/ClientApp/src/app/sys/services/tax-code-details.service.ts
import { Injectable } from '@angular/core';
import { HttpClient, HttpHeaders } from '@angular/common/http';
import { AppConfigService } from '../../app-config.service';


@Injectable({
  providedIn: 'root'
})
export class TaxCodeDetailsService {
  private url: string = "";
  private baseurl: string = "";

  constructor(public http: HttpClient, private appConfigService: AppConfigService) {
    this.baseurl = appConfigService.getServiceBaseUrl();
  }
  getAllTaxCodeDetails() {
    this.url = this.baseurl + 'api/SYS/TaxCodeDetails/GetAllTaxCodeDetails';
    return this.http.get(this.url);
  }
  addTaxCodeDetails(data) {
    this.url = this.baseurl + 'api/SYS/TaxCodeDetails/AddTaxCodeDetails';
    return this.http.post(this.url, data);
  }
  updateTaxCodeDetails(data) {
    this.url = this.baseurl + 'api/SYS/TaxCodeDetails/UpdateTaxCodeDetails';
    return this.http.post(this.url, data);
  }
  getTaxCodeDetailsByID(ID) {
    this.url = this.baseurl + 'api/SYS/TaxCodeDetails/GetTaxCodeDetailsByID/' + ID;
    return this.http.get(this.url);
  }
  getTaxCodeDetailsByTaxCodeID(ID) {
    this.url = this.baseurl + 'api/SYS/TaxCodeDetails/GetTaxCodeDetailsByTaxCodeID/' + ID;
    return this.http.get(this.url);
  }
  AddupdateWorkFlowDetails(data){
    debugger;
    this.url = this.baseurl + 'api/SYS/TaxCodeDetails/AddUpdateTaxcodeData';
    return this.http.post(this.url, data);
  }
}

File: LinkERP.BLL/SYS/TaxCodeDetailsService.cs
using LinkERP.BLL.SYS.Interfaces;
using LinkERP.DAL.SYS.Interfaces;
using LinkERP.Entity.SYS;
using System;
using System.Collections.Generic;
using System.Text;


namespace LinkERP.BLL.SYS
{
    public class TaxCodeDetailsService :ITaxCodeDetailsService
    {
        ITaxCodeDetailsRepository taxCodeDetails ;
        public TaxCodeDetailsService(ITaxCodeDetailsRepository _taxCodeDetails)
        {
            taxCodeDetails = _taxCodeDetails;
        }

        public string AddTaxCodeDetails(LBS_SYS_TaxCodeDetail lBS_SYS_TaxCodeDetail)
        {
            return taxCodeDetails.AddTaxCodeDetails(lBS_SYS_TaxCodeDetail);
        }
        public string UpdateTaxCodeDetails(LBS_SYS_TaxCodeDetail lBS_SYS_TaxCodeDetail)
        {
            return taxCodeDetails.UpdateTaxCodeDetails(lBS_SYS_TaxCodeDetail);
        }
        public IList<LBS_SYS_TaxCodeDetail> GetAllTaxCodeDetails()
        {
            return taxCodeDetails.GetAllTaxCodeDetails();
        }
        public LBS_SYS_TaxCodeDetail GetTaxCodeDetailsByID(Guid ID)
        {
            return taxCodeDetails.GetTaxCodeDetailsByID(ID);
        }
        public IList<LBS_SYS_TaxCodeDetail> GetTaxCodeDetailsByTaxCodeID(Guid ID)
        {
            return taxCodeDetails.GetTaxCodeDetailsByTaxCodeID(ID);
        }
        //public LBS_SYS_TaxCodeDetail GetTaxCodeDetailsByTaxCodeID(Guid ID)
        //{
        //    return taxCodeDetails.GetTaxCodeDetailsByTaxCodeID(ID);
        //}
        public string AddUpdateTaxcodeData(string lBS_SYS_TaxCodeDetail)
        {
            return taxCodeDetails.AddUpdateTaxcodeData(lBS_SYS_TaxCodeDetail);
        }
    }
}
